common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -268,7 +309,261 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
+            // Pairwise cipher for RSN/WPA2 (default: use wpa_pairwise value)
+            //  rsn_pairwise=CCMP
+            std::string wpa_pairwise(""); // default to empty -> delete from hostapd config
+
+            // WPA pre-shared keys for WPA-PSK. This can be either entered as a 256-bit
+            // secret in hex format (64 hex digits), wpa_psk, or as an ASCII passphrase
+            // (8..63 characters), wpa_passphrase.
+            std::string wpa_passphrase("");
+            std::string wpa_psk("");
+
+            // ieee80211w: Whether management frame protection (MFP) is enabled
+            // 0 = disabled (default)
+            // 1 = optional
+            // 2 = required
+            std::string ieee80211w("");
+
+            // This parameter can be used to disable caching of PMKSA created through EAP
+            // authentication. RSN preauthentication may still end up using PMKSA caching if
+            // it is enabled (rsn_preauth=1).
+            // 0 = PMKSA caching enabled (default)
+            // 1 = PMKSA caching disabled
+            std::string disable_pmksa_caching("");
+
+            // Opportunistic Key Caching (aka Proactive Key Caching)
+            // Allow PMK cache to be shared opportunistically among configured interfaces
+            // and BSSes (i.e., all configurations within a single hostapd process).
+            // 0 = disabled (default)
+            // 1 = enabled
+            std::string okc("");
+
+            // This parameter can be used to disable retransmission of EAPOL-Key frames that
+            // are used to install keys (EAPOL-Key message 3/4 and group message 1/2). This
+            // is similar to setting wpa_group_update_count=1 and
+            std::string wpa_disable_eapol_key_retries("");
+
+            // EasyMesh R1 only allows Open and WPA2 PSK auth&encryption methods.
+            // Quote: A Multi-AP Controller shall set the Authentication Type attribute
+            //        in M2 to indicate WPA2-Personal or Open System Authentication.
+            // bss_it->authentication_type is a bitfield, but we are not going
+            // to accept any combinations due to the above limitation.
+            if (bss_it->authentication_type == WSC::eWscAuth::WSC_AUTH_OPEN) {
+                wpa = 0x0;
+                if (bss_it->encryption_type != WSC::eWscEncr::WSC_ENCR_NONE) {
+                    LOG(ERROR) << "Autoconfiguration: " << vap << " encryption set on open VAP";
+                    abort = true;
+                    return;
+                }
+                if (bss_it->network_key.length() > 0) {
+                    LOG(ERROR) << "Autoconfiguration: " << vap << " network key set for open VAP";
+                    abort = true;
+                    return;
+                }
+            } else if (bss_it->authentication_type == WSC::eWscAuth::WSC_AUTH_WPA2PSK) {
+                wpa = 0x2;
+                wpa_key_mgmt.assign("WPA-PSK");
```

Comment:
Any particular reason to prefer this over
```suggestion
                wpa_key_mgmt = "WPA-PSK";
```
?

(If yes, we probably want a coding style update for it)

Reply:
They are the same: [No 3](https://en.cppreference.com/w/cpp/string/basic_string/operator%3D), [No 6](https://en.cppreference.com/w/cpp/string/basic_string/assign).
However, the entire function uses `assign`, so either I'll replace all, or none.